protocol/src/main/kotlin/models/Message.kt

```diff
@@ -2,88 +2,83 @@ package models
 
 import Mapper
 import com.fasterxml.jackson.annotation.JsonFormat
+import com.fasterxml.jackson.module.kotlin.readValue
 import dateTimeFormat
 import typeid.TypeID
 import java.time.OffsetDateTime
 
 // TODO: linter gonna yell at us for this, but I want the typeid and serialization to be ez for now
 enum class MessageKind {
-    rfq, order, orderstatus
+  rfq, quote, close, order, orderstatus
 }
 
 class MessageMetadata(
-    val kind: MessageKind,
-    val to: String,
-    val from: String,
-    val id: TypeID,
-    val exchangeId: TypeID,
-    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = dateTimeFormat, timezone = "UTC")
-    val createdAt: OffsetDateTime
+  val kind: MessageKind,
+  val to: String,
+  val from: String,
+  val id: TypeID,
+  val exchangeId: TypeID,
+  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = dateTimeFormat, timezone = "UTC")
+  val createdAt: OffsetDateTime
 )
 
-sealed interface MessageData
-
-abstract class Message<T : Any>(
-    val metadata: MessageMetadata,
-    val data: T,
-    var signature: String? = null
-) {
-    init {
-        when (metadata.kind) {
-            MessageKind.rfq -> require(data is RfqData)
-            MessageKind.order -> require(data is OrderData)
-            MessageKind.orderstatus -> require(data is OrderStatusData)
-        }
-
-        if (signature != null) {
-            verify()
-        } else {
-            validate()
-        }
-    }
 
-    private fun verify() {
-        validate()
+sealed class Message {
```

Comment:
Should this belong in the web5 repo? Seems related to DWNs, and not specifically to tbdex.

Reply:
@andresuribe87 this is actually a tbdex message as described in the tbdex protocol [here](https://github.com/TBD54566975/tbdex/tree/main/specs/protocol#messages)

Reply:
Thanks @ethan-tbd - is that the right place long term for it?